### PR TITLE
Make the MessageBodyPart struct deserializable from local cache

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: google-apis-common/src/serde.rs [hunk 1]
```diff
@@ -160,8 +162,8 @@ pub mod urlsafe_base64 {
         where
             D: Deserializer<'de>,
         {
-            let s: &str = Deserialize::deserialize(deserializer)?;
-            base64::decode_config(s, base64::URL_SAFE).map_err(serde::de::Error::custom)
+            let s: Cow<'de, String> = Deserialize::deserialize(deserializer)?;
```
Comment (Byron) [inline google-apis-common/src/serde.rs:165]: Can you elaborate why this is necessary?

File: gen/gmail1/Cargo.toml [hunk 2]
```diff
@@ -23,15 +23,14 @@ mime = "^ 0.3.0"
 serde = { version = "^ 1.0", features = ["derive"] }
 serde_json = "^ 1.0"
 itertools = "^ 0.10"
-google-apis-common = { path = "../../google-apis-common", version = "5.0.1" }
-hyper = "^ 0.14"
+google-apis-common = { path = "../../google-apis-common", version = "6" }
```
Comment (Byron) [inline gen/gmail1/Cargo.toml:26]: Please remove all changes to this file as it is generated (and says `# DO NOT EDIT!` in its first line).
Also, please refrain from adding features to `hyper` as that is something the application will control, it's not up to the library to select features unless they are needed for it to function, which I believe isn't the case here.

As you see [in the manifest source file](https://github.com/Byron/google-apis-rs/blob/209d70bfede6a4d0a3d75064fbd2f17d395d0095/src/generator/templates/Cargo.toml.mako#L37), the library is already upgraded, thus the code needs to be regenerated and probably republished to make it come through.

**Review by Byron — CHANGES_REQUESTED** [review]
Thanks for contributing.

In order to merge, I will need to understand more though.

File: gen/gmail1/Cargo.toml (hunk 2 above)
Comment (Yatekii) [reply gen/gmail1/Cargo.toml:26]: Aw shoot, yes, I will remove it. This was the lazy hack to make it work locally :)

File: google-apis-common/src/serde.rs (hunk 1 above)
Comment (Yatekii) [reply google-apis-common/src/serde.rs:165]: So when you try to deserialize from a file reader, serde is not happy and says it expected an owned string.

**Comment by Yatekii** [comment]
Sorry for the low effort PR ... will fix :)

**Comment by Byron** [comment]
I am sorry, but think it's best to close this PR due to inactivity. Please feel free to reopen or resubmit.

Thanks for your understanding.

**Pull request closed without merging**